src/v/storage/segment_utils.cc

```diff
@@ -97,7 +97,7 @@ static inline ss::file wrap_handle(ss::file f, debug_sanitize_files debug) {
 }
 
 ss::future<ss::file> make_handle(
-  const std::filesystem::path path,
+  const std::filesystem::path& path,
```

Comment:
If I'm reading this correctly, this will cause a UAF error:

```c++
# src/v/storage/snapshot.cc
auto path = _dir / filename;
return internal::make_handle(path, flags, {}, {}) ...

# internal::make_handle
ss::future<ss::file> make_handle(
const std::filesystem::path& path,      // <-- Reference to variable on stack (above)
    auto file = co_await ss::open_file_dma(path.string(), flags, opt);
    co_await maybe_disable_cow(path, file); // <-- UAF
```

because the caller isn't a coroutine and has path on the stack.

Reply:
fixed